macaroonbakery/tests/test_codec.py

```diff
@@ -24,13 +29,146 @@ def test_v1_round_trip(self):
+            'WQiOiJra0ZuOGJEaEt4RUxtUjd0NkJxTU0vdHhMMFVqaEZjR1'
+            'BORldUUExGdjVla1dWUjA4Uk1sbGJhc3c4VGdFbkhzM0laeVo'
+            '0V2lEOHhRUWdjU3ljOHY4eUt4dEhxejVEczJOYmh1ZDJhUFdt'
+            'UTVMcVlNWitmZ2FNaTAxdE9DIn0=')
+        cav = codec.decode_caveat(tp_key, encrypted_cav)
+        self.assertEquals(cav, macaroon.ThirdPartyCaveatInfo(
+            condition='caveat condition',
+            first_party_public_key=fp_key.public_key,
+            third_party_key_pair=tp_key,
+            root_key=b'random',
+            caveat=encrypted_cav,
+            version=bakery.BAKERY_V1,
+            ns=macaroon.legacy_namespace()
+        ))
+
+    def test_decode_caveat_v2_from_go(self):
+        tp_key = PrivateKey(base64.b64decode(
+            'TSpvLpQkRj+T3JXnsW2n43n5zP/0X4zn0RvDiWC3IJ0='))
+        fp_key = PrivateKey(base64.b64decode(
+            'KXpsoJ9ujZYi/O2Cca6kaWh65MSawzy79LWkrjOfzcs='))
+        # This caveat has been generated from the go code
+        # to check the compatibilty
+        encrypted_cav = base64.urlsafe_b64decode(
+            utils.add_base64_padding(six.b(
+                'AvD-xlUf2MdGMgtu7OKRQnCP1OQJk6PKeFWRK26WIBA6DNwKGIHq9xGcHS9IZ'
+                'Lh0cL6D9qpeKI0mXmCPfnwRQDuVYC8y5gVWd-oCGZaj5TGtk3byp2Vnw6ojmt'
+                'sULDhY59YA_J_Y0ATkERO5T9ajoRWBxU2OXBoX6bImXA')))
+        cav = codec.decode_caveat(tp_key, encrypted_cav)
+        self.assertEquals(cav, macaroon.ThirdPartyCaveatInfo(
+            condition='third party condition',
+            first_party_public_key=fp_key.public_key,
+            third_party_key_pair=tp_key,
+            root_key=b'random',
+            caveat=encrypted_cav,
+            version=bakery.BAKERY_V2,
+            ns=macaroon.legacy_namespace()
+        ))
+
+    def test_decode_caveat_v3_from_go(self):
+        tp_key = PrivateKey(base64.b64decode(
+            'TSpvLpQkRj+T3JXnsW2n43n5zP/0X4zn0RvDiWC3IJ0='))
+        fp_key = PrivateKey(base64.b64decode(
+            'KXpsoJ9ujZYi/O2Cca6kaWh65MSawzy79LWkrjOfzcs='))
+        # This caveat has been generated from the go code
+        # to check the compatibilty
+        encrypted_cav = base64.urlsafe_b64decode(
+            utils.add_base64_padding(six.b(
+                'A_D-xlUf2MdGMgtu7OKRQnCP1OQJk6PKeFWRK26WIBA6DNwKGNLeFSkD2M-8A'
+                'EYvmgVH95GWu7T7caKxKhhOQFcEKgnXKJvYXxz1zin4cZc4Q6C7gVqA-J4_j3'
+                '1LX4VKxymqG62UGPo78wOv0_fKjr3OI6PPJOYOQgBMclemlRF2')))
+        cav = codec.decode_caveat(tp_key, encrypted_cav)
+        self.assertEquals(cav, macaroon.ThirdPartyCaveatInfo(
+            condition='third party condition',
+            first_party_public_key=fp_key.public_key,
+            third_party_key_pair=tp_key,
+            root_key=b'random',
+            caveat=encrypted_cav,
+            version=bakery.BAKERY_V3,
+            ns=macaroon.legacy_namespace()
+        ))
+
+    def test_encode_decode_varint(self):
```

Comment:
These tests aren't sufficient. They test the round trip but don't check that the data is represented as expected (which is required if this is to work correctly.)